internal/service/api/handlers/get_gist_data.go

```diff
@@ -25,72 +20,21 @@ func GetGistData(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	userDID, err := w3c.ParseDID(req.UserDID)
-	if err != nil {
-		Log(r).WithError(err).Error("failed to parse user DID")
-		ape.RenderErr(w, problems.BadRequest(err)...)
-		return
-	}
-
-	userID, err := core.IDFromDID(*userDID)
+	userID, err := core.IDFromDID(*req.UserDID)
 	if err != nil {
 		Log(r).WithError(err).Error("failed to parse user ID")
 		ape.RenderErr(w, problems.InternalError())
 		return
 	}
 
-	blockNum, err := EthClient(r).BlockNumber(context.Background())
-	if err != nil {
-		Log(r).WithError(err).Error("failed to get block number")
-		ape.RenderErr(w, problems.InternalError())
-		return
-	}
-
-	if req.BlockNumber > blockNum {
-		Log(r).WithFields(logan.F{
-			"requested_block_number": req.BlockNumber,
-			"latest_block_number":    blockNum,
-		}).Error("Requested block number is higher than latest")
-		ape.RenderErr(w, problems.BadRequest(validation.Errors{
-			"/block_number": errors.New("Requested block number is higher than latest"),
-		})...)
-		return
-	}
-
-	if req.BlockNumber != 0 {
-		blockNum = req.BlockNumber
-	}
-
-	stateContract := StateContract(r)
-
-	gistProof, err := stateContract.GetGISTProof(&bind.CallOpts{
-		BlockNumber: new(big.Int).SetUint64(blockNum),
-	}, userID.BigInt())
+	gistProof, err := StateContract(r).GetGISTProofByRoot(&bind.CallOpts{}, req.StateRoot, userID.BigInt())
```

Comment:
This is wrong

Actual contract function signature: 

```solidity
    function getGISTProofByRoot(
        uint256 id_,
        uint256 root_
    ) external view returns (IState.GistProof memory) {
        return _smtProofAdapter(_gistData.getProofByRoot(PoseidonUnit1L.poseidon([id_]), root_));
    }
 ```
 
 ```suggestion
	gistProof, err := StateContract(r).GetGISTProofByRoot(&bind.CallOpts{}, userID.BigInt(), req.StateRoot)
```